starters/basic-starter/components/meta.tsx

```diff
@@ -0,0 +1,113 @@
+import { DrupalMetatag } from 'types/drupal';
+import { imageAbsoluteUrl } from '../lib/image-absolute-url';
+
+interface MetaProps {
+  title?: string;
+  path?: string;
+  tags?: DrupalMetatag[];
+}
+
+export function Meta({ tags }: MetaProps) {
+  const clientSideUrl =
+    typeof window !== 'undefined' && window.location.origin
+      ? window.location.origin
+      : '';
+
+  const router = useRouter();
+  const siteUrl = process.env.NEXT_PUBLIC_SITE_BASE_URL
+    ? process.env.NEXT_PUBLIC_SITE_BASE_URL
+    : clientSideUrl;
+
+  const currentUrl = `${siteUrl}${router.asPath !== '/' ? router.asPath : ''}`;
+  const absoluteUrl = currentUrl.split('?')[0];
+  // Initialize imageUrl variable with default value.
+  const imageUrl = `${currentUrl}${process.env.imagePath}?url=`;
+  // Initialize schemaMetatag variable with empty object.
+  const schemaMetatag = {};
+  // Initialize schemaMetatagData variable with empty object.
+  const schemaMetatagData = {};
+  // Check whether the tags is an array list.
+  if (Array.isArray(tags)) {
+    tags = tags.filter((obj) => {
+      // Set og:url to page url.
+      if (obj.attributes.property === 'og:url') {
+        obj.attributes.content = absoluteUrl;
+      }
+      // Return null if keywords is empty.
+      if (obj.attributes.name === 'keywords') {
+        obj.attributes.content = obj.attributes.content.replace(/,\s*$/, '');
+      }
+
+      // Prepare schema metatag object.
+      if (obj.attributes.schema_metatag) {
+        schemaMetatag[obj.attributes.name] =
+          obj.attributes.name === 'image'
+            ? imageAbsoluteUrl(imageUrl, obj.attributes.content.url)
+            : obj.attributes.content;
+      }
+      // Canonical and schem_metatag both needs to render inside link and
+      // script tag respectively and if any attributes having no content then,
+      // we are returning null from here.
+      if (
+        obj.attributes.rel === 'canonical' ||
+        obj.attributes.schema_metatag ||
+        obj.attributes.content.length == 0
+      ) {
+        return null;
+      }
+
+      return obj;
+    });
+
+    // Prepare schema metatag data object.
+    schemaMetatagData['@context'] = 'https://schema.org';
+    schemaMetatagData['@graph'] = [schemaMetatag];
+  }
+
+  return (
+    <Head>
+      <script type="application/ld+json">
+        {JSON.stringify(schemaMetatagData, null, 2)}
+      </script>
+      <link rel="canonical" href={absoluteUrl} />
+      {tags?.length ? (
+        tags.map((tag, index) => {
+          if (tag.attributes.name === 'title') {
+            return (
+              <title key={tag.attributes.name}>{tag.attributes.content}</title>
+            );
+          }
+          if (tag.attributes.property === 'og:image') {
+            return (
+              <meta
+                key={index}
+                property={tag.attributes.property}
+                content={imageAbsoluteUrl(imageUrl, tag.attributes.content)}
```

Comment:
This is not working as it should. I'm getting following output rendered on the server side if I go to `http://localhost:3000/article/blog/article-ten-medium-length-placeholder-heading?123#test`:

```
<meta property="og:image" content="/article/blog/article-ten-medium-length-placeholder-heading?123=/_next/image?url=http%3A%2F%2Facms.ddev.site%2Fsites%2Fdefault%2Ffiles%2F2021-02%2FImage-placeholder_1.png&amp;w=1200&amp;q=75">
```